File: benchmark/maps.py
from __future__ import print_function

import time

map = {}

start = time.clock()

for i in range(1, 100001):
  map[i] = i

sum = 0
for i in range(1, 100001):
  sum = sum + map[i]
print(sum)

for i in range(1, 100001):
  del map[i]

print("elapsed: " + str(time.clock() - start))

File: benchmark/maps.rb
map = Hash.new

start = Time.now

for i in (1..100000)
  map[i] = i
end

sum = 0
for i in (1..100000)
  sum = sum + map[i]
end
puts sum

for i in (1..100000)
  map.delete(i)
end

puts "elapsed: " + (Time.now - start).to_s
